syntax = "proto3";

package vio_input_proto;

message Vec3 {
  double x = 1;
  double y = 2;
  double z = 3;
}

message IMUCamData
{
  fixed64 timestamp = 1;
  Vec3 angular_vel = 2;
  Vec3 linear_accel = 3;

  int32 rows = 4;
  int32 cols = 5;
  bytes img0_data = 6;
  bytes img1_data = 7;
}

message IMUCamVec {
  repeated IMUCamData imu_cam_data = 1;
}